proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```diff
@@ -964,6 +965,26 @@ void setClientBrand(String clientBrand) {
     this.clientBrand = clientBrand;
   }
 
+  @Override
+  public void addCustomChatCompletions(@NotNull Collection<String> completions) {
+    this.sendCustomChatCompletionPacket(completions, PlayerChatCompletionPacket.Action.ADD);
+  }
+
+  @Override
+  public void removeCustomChatCompletions(@NotNull Collection<String> completions) {
+    this.sendCustomChatCompletionPacket(completions, PlayerChatCompletionPacket.Action.REMOVE);
+  }
+
+  @Override
+  public void setCustomChatCompletions(@NotNull Collection<String> completions) {
+    this.sendCustomChatCompletionPacket(completions, PlayerChatCompletionPacket.Action.SET);
+  }
+
+  private void sendCustomChatCompletionPacket(@NotNull Collection<String> completions,
+                                              PlayerChatCompletionPacket.Action action) {
+    connection.write(new PlayerChatCompletionPacket(completions.toArray(new String[0]), action));
+  }
```

Comment:
You should check if the player has version 1.19 or higher since in previous versions this feature did not exist

Reply:
Also, you must add null checks

Reply:
Oh right, I'm spoiled from writing mostly Kotlin these days :D 

What's the course of action, just drop if the player is below 1.19?